server/unenv.go

```diff
@@ -0,0 +1,146 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	esbuild "github.com/evanw/esbuild/pkg/api"
+)
+
+const (
+	// https://github.com/unjs/unenv
+	unenvVersion = "e666fcf"
+)
+
+var (
+	unenvNodeRuntimeBulid = map[string][]byte{}
+)
+
+func buildUnenvNodeRuntime() (err error) {
+	wd := path.Join(config.WorkDir, "npm/gh/unjs/unenv@"+unenvVersion)
+	err = ensureDir(wd)
+	if err != nil {
+		return err
+	}
+
+	// ensure 'package.json' file to prevent read up-levels
+	packageJsonFp := path.Join(wd, "package.json")
+	if !existsFile(packageJsonFp) {
+		err = os.WriteFile(packageJsonFp, []byte(`{"dependencies":{"unenv":"github:unjs/unenv#`+unenvVersion+`"}}`), 0644)
+		if err != nil {
+			return
+		}
+	}
+
+	cmd := exec.Command("pnpm", "i", "--prefer-offline")
+	cmd.Dir = wd
+	err = cmd.Run()
+	if err != nil {
+		err = fmt.Errorf("install unjs/unenv from github: %v", err)
+		return
+	}
+
+	endpoints := make([]esbuild.EntryPoint, 0, len(nodeBuiltinModules))
+	for name := range nodeBuiltinModules {
+		filename := path.Join(wd, "node_modules", "unenv/runtime/node/"+name+"/index.mjs")
+		if existsFile(filename) {
+			endpoints = append(endpoints, esbuild.EntryPoint{
+				InputPath:  filename,
+				OutputPath: name,
+			})
+		}
+	}
+
+	ret := esbuild.Build(esbuild.BuildOptions{
+		AbsWorkingDir:       wd,
+		EntryPointsAdvanced: endpoints,
+		Platform:            esbuild.PlatformBrowser,
+		Format:              esbuild.FormatESModule,
+		Target:              esbuild.ESNext,
+		Bundle:              true,
+		Splitting:           true,
+		MinifyWhitespace:    true,
+		MinifyIdentifiers:   true,
+		MinifySyntax:        true,
+		OutExtension:        map[string]string{".js": ".mjs"},
+		Write:               false,
+		Outdir:              "/",
+		Plugins: []esbuild.Plugin{
+			{
+				Name: "unenv",
```

Comment:
@pi0 thanks for the great design of unenv, i just need to add few code to make it work 🤩